### PR TITLE
chore: [ADR-0001] use architecture decision records

### LINKED ISSUE #676: Create a Markdown template for the ADR
I'd suggest we start with something like this: https://github.com/joelparkerhenderson/architecture-decision-record/blob/main/templates/decision-record-template-by-michael-nygard/index.md.

Related to #663

### PR DESCRIPTION
**Pull request by Amzani**
**Description**

Start using Michael Nygard ADR template.

**Related issue(s)**

Resolves #676

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,3 +50,14 @@ npm run build:ds
 npm run build
 ```
 
+## Architecture decision records
+
+### Create a new architecture decision record
+
+- Install [adr tools](https://github.com/adr/adr-tools/blob/patch-1/INSTALL.md)
```
Comment (fmvilas) [inline README.md:57]: This repository has been archived in 2022. I think we should probably not use this tool.

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Amzani) [reply README.md:57]: @fmvilas most of adr tools I find in the OSS are outdated I'm going to build a small one that implement this template.

Comment (fmvilas) [reply README.md:57]: I think a manual approach is fine. It's a really simple template :)

File: doc/adr/0001-record-architecture-decisions.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+Date: 2023-06-27
+
+## Status
+
+Accepted
+
+## Context
+
+We need to record the architectural decisions made on this project.
```
Comment (fmvilas) [inline doc/adr/0001-record-architecture-decisions.md:11]: Can you expand on the why here? You can get it directly from https://github.com/asyncapi/studio/issues/663 if you don't want to scratch your head 😄

**Review by fmvilas — CHANGES_REQUESTED** [review]

**Review by fmvilas — APPROVED** [review]

**Pull request merged**